pandas/core/arrays/categorical.py

```diff
@@ -2974,8 +2974,7 @@ def _delegate_method(self, name: str, *args, **kwargs):
         from pandas import Series
 
         method = getattr(self._parent, name)
-        res = method(*args, **kwargs)
-        if res is not None:
+        if (res := method(*args, **kwargs)) is not None:
             return Series(res, index=self._index, name=self._name)
```

Comment:
These are no more readable (honestly they seem even be less readable) and don't impact behavior or performance. Seems like needless code churn to me, but if others really want this no strong objection.

Reply:
that's fair, I'll revert 👍